src/components/GotoLineModal.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { getActiveSearch, getSelectedSource } from "../selectors";
+import actions from "../actions";
+
+import Modal from "./shared/Modal";
+
+import SearchInput from "./shared/SearchInput";
+
+import type { SourceRecord } from "../reducers/sources";
+import type { SelectSourceOptions } from "../actions/sources";
+
+type GotoLineModalState = {
+  query: ?string
+};
+
+import "./SymbolModal.css";
+
+class GotoLineModal extends Component {
+  state: GotoLineModalState;
+
+  props: {
+    enabled: boolean,
+    selectSource: (string, ?SelectSourceOptions) => void,
+    selectedSource?: SourceRecord,
+    closeActiveSearch: () => void,
+    highlightLineRange: ({ start: number, end: number }) => void,
+    clearHighlightLineRange: () => void
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { query: "" };
+  }
+
+  onClick = (e: SyntheticEvent) => {
+    e.stopPropagation();
+  };
+
+  onChange = (e: SyntheticInputEvent) => {
+    const { selectedSource } = this.props;
+    if (!selectedSource || !selectedSource.get("text")) {
+      return;
+    }
+
+    this.setState({ query: e.target.value });
+  };
+
+  closeModal = () => {
+    this.props.closeActiveSearch();
+    this.props.clearHighlightLineRange();
+  };
+
+  onKeyUp = (e: SyntheticKeyboardEvent) => {
+    e.preventDefault();
+    const { selectSource, selectedSource, enabled } = this.props;
+    const { query } = this.state;
+
+    if (!enabled || !selectedSource) {
+      return;
+    }
+
+    if (e.key === "Enter" && query != null) {
+      const linenumber = parseInt(query.replace(/[^\d+]/g, ""), 10);
+      if (!isNaN(linenumber)) {
+        selectSource(selectedSource.get("id"), { line: linenumber });
+      }
+      this.closeModal();
+    } else if (e.key === "Tab") {
+      this.closeModal();
+    }
```

Comment:
I think we might want to handle "Escape" too

Reply:
Escape already closes out the modal in both SymbolModal and GotoLineModal, it's handled more globally [here](https://github.com/devtools-html/debugger.html/blob/master/src/actions/ui.js#L7) via [here](https://github.com/devtools-html/debugger.html/blob/master/src/components/App.js#L118) and [here](https://github.com/devtools-html/debugger.html/blob/master/src/components/App.js#L105).